### ISSUE TITLE
  nova salt-cloud -P  Private IPs returned, but not public. Checking for misidentified IPs

### ISSUE THREAD

**Issue by stamak** [saltstack/salt#42644]
### Description of Issue/Question

Private IPs returned, but not public. Checking for misidentified IPs

salt-cloud driver nova:
`` salt-cloud -P -m multiple_vms.map``  assigns floating ip for ONLY for one VM
for others rises such kind of error 
```
[WARNING ] Private IPs returned, but not public. Checking for misidentified IPs
[WARNING ] 10.64.137.28 is a private IP
```

without `` -P ``, (e.g. `` salt-cloud  -m multiple_vms.map ``) -  all is great 

### Setup

provider
```
os-my-proj-provider:
  driver: nova
  insecure: True
  use_keystoneauth: False
  auth_version: 2
  identity_url: https://xxx.xxx.xxx.xxx:5000/v2.0
  compute_region: RegionOne
  compute_name: nova
  tenant: my-proj
  user: my-proj-admin
  password: mypass
  ssh_key_name: my-proj
  ssh_key_file: /etc/salt/pki/os-my-proj-ssh-key.pem
  ssh_connect_timeout: 120
  wait_for_passwd_timeout: 200
  wait_for_passwd_maxtries: 500
```

profile
```
os-my-proj-ubuntu-16.04-m1-small:                                                                                        [84/876]
  provider: os-my-proj-provider
  size: m1.small
  image: ubuntu-16.04
  ssh_username: ubuntu
  networks:
    - net-id: 26871ef8-3454-98e7-86eafb9cf73b
  floating_ip:
    auto_assign: True
    pool: my_floating_net
```

map:
```
os-my-proj-ubuntu-16.04-m1-large:
  - my-node2-my-proj.demo:
  - my-node3-my-proj.demo:
  - my-node1-my-proj.demo:

```
### Versions Report
```
salt-master --versions-report                                                                      [10/728]
Salt Version:
           Salt: 2017.7.0

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.4.2
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 1.0.3
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.2.0
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: Ubuntu 16.04 xenial
         locale: UTF-8
        machine: x86_64
        release: 4.4.0-87-generic
         system: Linux
        version: Ubuntu 16.04 xenial

```

**Comment by rickh563** [comment]
ZD-1641

**Comment by UtahDave** [comment]
Is this a duplicate of https://github.com/saltstack/salt/pull/42253 ?

**Comment by gtmanfred** [comment]
the fix is for the openstack driver, this issue is for the nova driver.

**Comment by stamak** [comment]
@gtmanfred the problem is still exists on 2016.11.6
Was it backported to 2016.11.6?

**Comment by gtmanfred** [comment]
It was fixed on the `openstack` driver, not the `nova` driver.  But it looks like a similar bug was introduced for the `nova` driver


For the map, does only 1 public ip get allocated for all the servers in the map when building in parallel?

**Comment by samodid** [comment]
Yes, it allocates only 1 public ip

**Comment by gtmanfred** [comment]
This should be fixed by #42940 I am working on testing it now.

Daniel
